assets/src/design-system/testUtils/renderWithProviders.js

```diff
@@ -18,17 +18,33 @@
  * External dependencies
  */
 import { render } from '@testing-library/react';
+import PropTypes from 'prop-types';
 import { ThemeProvider } from 'styled-components';
 
 /**
  * Internal dependencies
  */
 import { lightMode, theme } from '../theme';
 
-export const renderWithProviders = (children) => {
-  return render(
+const ProviderWrapper = ({ children }) => {
+  return (
     <ThemeProvider theme={{ ...theme, colors: { ...lightMode } }}>
       {children}
     </ThemeProvider>
   );
 };
+ProviderWrapper.propTypes = {
+  children: PropTypes.node,
+};
+
+export const renderWithProviders = (children) => {
+  const { rerender, ...result } = render(
+    <ProviderWrapper>{children}</ProviderWrapper>
+  );
+  return {
+    ...result,
+    // rerender needs to be wrapped in the providers too
+    rerender: (rerenderedChildren) =>
+      rerender(<ProviderWrapper>{rerenderedChildren}</ProviderWrapper>),
+  };
+};
```

Comment:
When calling `rerender` we weren't passing the providers again. Needed to add a wrapper around the `rerender` function exposed through `renderWithProviders`.

Reply:
This should not be needed.

I think what you need to do is something like this:

```js
const renderWithProviders = (ui, options) =>
  render(ui, {
    wrapper: WithThemeProvider,
    queries: {
      ...queries,
      ...ariaLabelQueries,
      ...autoAdvanceAfterQueries,
      ...idQueries,
    },
    ...options,
  });
```

See `renderWithTheme` in the editor for comparison.

https://testing-library.com/docs/react-testing-library/setup#custom-render